nmostesting/ControllerTest.py

```diff
@@ -0,0 +1,530 @@
+        device_data["senders"] = [] 
+        device_data["receivers"] = [ receiver["id"] ] 
+        self.post_resource(self, "device", device_data, codes=codes, fail=fail)
+
+        # Register receiver
+        receiver_data = self._create_receiver_json(receiver)
+
+        self.post_resource(self, "receiver", receiver_data, codes=codes, fail=fail)
+
+    def _delete_receiver(self, test, receiver):
+        
+        del_url = self.mock_registry_base_url + 'x-nmos/registration/v1.3/resource/receivers/' + receiver['id']
+        
+        valid, r = self.do_request("DELETE", del_url)
+        if not valid:
+            # Hmm - do we need these exceptions as the registry is our own mock registry?
+            raise NMOSTestException(test.FAIL(test, "Registration API returned an unexpected response: {}".format(r)))
+
+        del_url = self.mock_registry_base_url + 'x-nmos/registration/v1.3/resource/devices/' + receiver['device_id']
+        
+        valid, r = self.do_request("DELETE", del_url)
+        if not valid:
+            # Hmm - do we need these exceptions as the registry is our own mock registry?
+            raise NMOSTestException(test.FAIL(test, "Registration API returned an unexpected response: {}".format(r)))
+
+    def pre_tests_message(self):
+        """
+        Introduction to NMOS Controller Test Suite
+        """
+        dns_sd_enabled = CONFIG.ENABLE_DNS_SD and CONFIG.DNS_SD_MODE == "unicast"
+
+        paragraphs = []
+        paragraphs.append('These tests validate an NMOS Controller under Test (NCuT).\n\n')
+
+        paragraphs.append('A Test AMWA IS-04 reference Registry is available on the network')
+
+        paragraphs.append(' and is being advertised via unicast DNS-SD.\n\n' if dns_sd_enabled else '.\n\n')
+
+        paragraphs.append('Please ensure that the following configuration has been set on the NCuT machine.\n\n ' \
+            '* Ensure that the primary DNS of the NCuT machine has been set to \"' + get_default_ip() + '\". \n' \
+            '* Ensure that the NCuT unicast search domain is set to \"' + CONFIG.DNS_DOMAIN + '\". \n\n' \
+            'Alternatively it '  if dns_sd_enabled else 'It ')
+        
+        paragraphs.append('is possible to reach the Registry via the following URL:\n\n' + self.mock_registry_base_url + 'x-nmos/query/v1.3\n\n'\
+            'Please ensure the NCuT has located the test AMWA IS-04 Registry before clicking the \'Next\' button.')
+
+        question = ''.join(paragraphs)
+                
+        try:
+            self._invoke_testing_facade(question, [], test_type="action", timeout=600)
+
+        except TestingFacadeException as e:
+            # pre_test_introducton timed out
+            pass
+
+    def post_tests_message(self):
+        """
+        NMOS Controller Test Suite complete!
+        """
+        question =  'NMOS Controller Test Suite complete!\r\n\r\nPlease press the \'Next\' button to exit the tests'
```

Comment:
It might be possible to avoid the explicit escape sequences using multi-line strings, but I haven't tested this in the way these are used:
```suggestion
        question =  """NMOS Controller Test Suite complete!
        
                     Please press the 'Next' button to exit the tests"""
```

Reply:
I'll give this a try as it is definitely more elegant to look at

Reply:
Multi-line strings now employed in all appropriate places